hydra-node/src/Hydra/Chain/Blockfrost.hs

```diff
@@ -65,3 +118,175 @@ mkTinyWallet tracer config = do
   toEpochInfo (EraHistory interpreter) =
     hoistEpochInfo (first show . runExcept) $
       Consensus.interpreterToEpochInfo interpreter
+
+blockfrostChain ::
+  (MonadIO m, MonadCatch m, MonadAsync m, MonadDelay m) =>
+  Tracer m DirectChainLog ->
+  TQueue m (Tx, TMVar m (Maybe (PostTxError Tx))) ->
```

Comment:
This is taken a bit out of context (from `Hydra.Chain.Direct`) and will make hard to understand code. Instead we should aim for an abstraction where the blockfrost chain backend has a means to wait for next tx to submit (and return success or an error). The fact that this is a queue should not leak all over the place.

Reply:
I don't quite follow you on this one...anyway, now we have `withBlockfrostChain` so everything is encapsulated in there.